dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -447,41 +447,59 @@ public static class DMB {
             private DMB() {
             }
 
-            public static Fence newBarrier() {
+            public static GenericVisibleEvent newBarrier() {
                 return newSYBarrier(); // Default barrier
             }
 
-            public static Fence newSYBarrier() {
-                return new Fence("DMB.SY");
+            public static GenericVisibleEvent newSYBarrier() {
+                String name = "DMB.SY";
+                GenericVisibleEvent fence = newFence(name);
+                fence.addTags(name);
+                return fence;
             }
 
-            public static Fence newISHBarrier() {
-                return new Fence("DMB.ISH");
+            public static GenericVisibleEvent newISHBarrier() {
+                String name = "DMB.ISH";
+                GenericVisibleEvent fence = newFence(name);
+                fence.addTags(name);
+                return fence;
```

Comment:
If you fix the `newFence` method, you can reduce this back to `return newFence("DMB.ISH")`.
Same for all other fence methods